tiles/src/main/java/com/protomaps/basemap/layers/Places.java

```diff
@@ -38,31 +39,66 @@ public void processFeature(SourceFeature sf, FeatureCollector features) {
       } else if (sf.hasTag("place", "city")) {
         feat.setAttr("pmap:kind", "city")
           .setZoomRange(4, 15);
-
-        if (sf.getString("population") != null) {
-          Integer population = Parse.parseIntOrNull(sf.getString("population"));
-          if (population != null) {
-            feat.setAttr("population", population);
-            feat.setSortKey((int) Math.log(population));
-            // TODO: use label grid
-          } else {
-            feat.setSortKey(0);
-          }
+        if ( population.equals(0) ) {
+          population = 10000;
         }
-
-      } else if (sf.hasTag("place", "suburb")) {
-        feat.setAttr("pmap:kind", "neighbourhood")
-          .setZoomRange(8, 15);
       } else if (sf.hasTag("place", "town")) {
         feat.setAttr("pmap:kind", "neighbourhood")
           .setZoomRange(8, 15);
+        if ( population.equals(0) ) {
+          population = 5000;
+        }
       } else if (sf.hasTag("place", "village")) {
         feat.setAttr("pmap:kind", "neighbourhood")
-          .setZoomRange(10, 15);
+                .setZoomRange(10, 15);
+        if ( population.equals(0) ) {
+          population = 2000;
+        }
+      } else if (sf.hasTag("place", "suburb")) {
+        feat.setAttr("pmap:kind", "neighbourhood")
+                .setZoomRange(8, 15);
       } else {
         feat.setAttr("pmap:kind", "neighbourhood")
           .setZoomRange(12, 15);
       }
+
+      if (population != null) {
+        feat.setAttr("population", population);
+        feat.setSortKey((int) Math.log(population));
+        // TODO: use label grid
+      } else {
+        feat.setSortKey(0);
+      }
+
+      Integer population_rank = 0;
+
+      int[] pop_breaks = {
+              1000000000,
+              100000000,
+              50000000,
+              20000000,
+              10000000,
+              5000000,
+              1000000,
+              500000,
+              200000,
+              100000,
+              50000,
+              20000,
+              10000,
+              5000,
+              2000,
+              1000,
+              200,
+              0};
+
+      for (int i = 0; i < pop_breaks.length; i++) {
```

Comment:
This isn't working in Java, but works fine in Python. Help!

Reply:
I think you want to `break` once you find a result in the array that matches, right?

it seems like `population_rank` at the top is assigned to 0 or the OSM value. Since there's not really a case where it's nullable I don't think you need to use the `Integer` type, just plain `int` should be OK.

Reply:
LOL, yes. Fixed via 5b62ac04e9191fe6169821beeca4c7ec87218ef0

<img width="241" alt="image" src="https://github.com/protomaps/basemaps/assets/853051/1661cded-d7d3-4846-9ea7-c557c5785a70">

<img width="252" alt="image" src="https://github.com/protomaps/basemaps/assets/853051/dc6a808f-f06a-4a98-b0be-68ab9cb82980">